source/_components/google_pubsub.markdown

```diff
@@ -0,0 +1,91 @@
+---
+layout: page
+title: "Google Pub/Sub"
+description: "Setup for Google Pub/Sub integration"
+date: 2019-01-12 13:30
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: google-pubsub.png
+ha_category: Component
+ha_release: 0.86
+---
+
+The `google_pubsub` component allows you to hook into the Home Assistant event bus and send events to [Google Cloud Pub/Sub](https://cloud.google.com/pubsub/docs/overview). The current [free tier](https://cloud.google.com/free/) of GCP should allow you to sync about 1 event every 2 seconds on average (2 million invocations per month).
+
+## {% linkable_title First time setup %}
+
+This assumes you already have a Google Cloud project. If you don't, please create one in the [Google Cloud Console](https://console.cloud.google.com/projectcreate)
+
+You need to create a Service Account key in the [Google Cloud API Console](https://console.cloud.google.com/apis/credentials/serviceaccountkey)
+- Choose a new "New Service Account", give it a name and leave the key type as JSON
+- Select the role: Pub/Sub Publisher 
+
+This will download the Service Account JSON key to your machine. Do NOT share this with anyone. Place this file in your Home Assistant config folder.
+
+Next, create a Google Pub/Sub topic in the [Google Cloud API Console](https://console.cloud.google.com/cloudpubsub/topicList). The topic name will become something like `projects/project-198373/topics/topic-name`. Note the last part only (the name you chose): `topic-name`.
+
+
+## {% linkable_title Configuration %}
+
+Add the following lines to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+google_pubsub:
+  project_id: YOUR_PROJECT_ID
+  topic_name: YOUR_TOPIC_NAME
+  credentials_json: CREDENTIALS_FILENAME
+    - switch
+    - light
+    - group
+  filter:
```

Comment:
We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

Reply:
Removed the `filter` from the example and added a note explaining why users should care about this setting 👍 .